### PR TITLE
Fix migration from GitBucket (#22477)

### PR DESCRIPTION
**Pull request by haruo31**
Backport #22477 

Migration from GitBucket does not work due to a access for "Reviews" API on GitBucket that makes 404 response.
This PR has following changes.
1. Made to stop access for Reviews API while migrating from GitBucket.
2. Added support for custom URL (e.g. `http://example.com/gitbucket/owner/repository`)
3. Made to accept for git checkout URL (`http://example.com/git/owner/repository.git`)

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: services/migrations/gitbucket.go [hunk 1]
```diff
@@ -34,10 +34,11 @@ func (f *GitBucketDownloaderFactory) New(ctx context.Context, opts base.MigrateO
 		return nil, err
 	}
 
-	baseURL := u.Scheme + "://" + u.Host
 	fields := strings.Split(u.Path, "/")
-	oldOwner := fields[1]
-	oldName := strings.TrimSuffix(fields[2], ".git")
+	baseURL := u.Scheme + "://" + u.Host + strings.TrimSuffix(strings.Join(fields[:len(fields)-2], "/"), "/git")
```
Comment (wxiaoguang) [inline services/migrations/gitbucket.go:41]: If a user provides an invalid URL, could the `fields[:len(fields)-2]` be incorrect  or make Gitea panic/500?

----

the old code was not strict before, so maybe there is no problem.

Comment (zeripath) [reply services/migrations/gitbucket.go:41]:
```suggestion
	if len(fields) < 2 {
		return nil, fmt.Errorf("invalid path: %s", u.Path)
	}
	baseURL := u.Scheme + "://" + u.Host + strings.TrimSuffix(strings.Join(fields[:len(fields)-2], "/"), "/git")
```

**Comment by wolfogre** [comment]
Sorry, I'm confused. Why does this PR try to merge to `release/v1.18`? It's not a backport, or the bug is gone in the `main` version?

**Comment by haruo31** [comment]
Thank you for your help!
A part of this pull request has fix for migration bug. So I thought it would fall under following role.
> 1. Make sure you are targeting the `main` branch, pull requests on release branches are only allowed for bug fixes.
However, I added extra changes in this request.
If it should be for main branch, I will change base branch to `main`.
Please let me know.

**Comment by wolfogre** [comment]
> Thank you for your help! A part of this pull request has fix for migration bug. So I thought it would fall under following role.
> 
> > 1. Make sure you are targeting the `main` branch, pull requests on release branches are only allowed for bug fixes.
> 
> However, I added extra changes in this request. If it should be for main branch, I will change base branch to `main`. Please let me know.

Let me show you an example:

This PR has fixed a bug:

`https://github.com/go-gitea/gitea/pull/22392`

You may notice it has `backport/1.18` label and is based on `main`.

When it has been merged, the commit has been picked up to `release/1.18` branch, and started a new PR to `release/1.18`:

`https://github.com/go-gitea/gitea/pull/22474`

Or you can just fix this bug in `main` branch and let maintainers do the rest. 😂

**Comment by lunny** [comment]
backport #22477

**Review by wolfogre — APPROVED** [review]

**Comment by lunny** [comment]
make LG-TM work

**Comment by zeripath** [comment]
Please send backport

**Pull request merged**